pkg/controller/che/che_controller.go

```diff
@@ -254,6 +253,34 @@ func (r *ReconcileChe) Reconcile(request reconcile.Request) (reconcile.Result, e
 			}
 		}
 	}
+
+	// Get custom ConfigMap
+	// if it exists, add the data into CustomCheProperties
+	customConfigMap := &corev1.ConfigMap{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Namespace: instance.Namespace, Name: "custom"}, customConfigMap)
+	if err != nil && !errors.IsNotFound(err) {
+		logrus.Errorf("Error getting custom configMap: %v", err)
+		return reconcile.Result{}, err
+	}
+	if err == nil {
+		logrus.Infof("Found legacy custom ConfigMap.  Adding those values to CheCluster.Spec.Server.CustomCheProperties")
+		if instance.Spec.Server.CustomCheProperties == nil {
+			instance.Spec.Server.CustomCheProperties = make(map[string]string)
+		}
+		for k, v := range customConfigMap.Data {
+			instance.Spec.Server.CustomCheProperties[k] = v
+		}
+		if err = r.client.Delete(context.TODO(), customConfigMap); err != nil {
```

Comment:
I think I would delete *after* having successfully updated the custom resource with the `customConfigMap`.
Else we have a risk to loose the content of the old `custom` config map if an error occurs during the update just below